### PR TITLE
Tap15.0.1 support

### PR DESCRIPTION
**Pull request by makuzaverite**
Migrate tests to tap15.0.1

#### Checklist

- [x] run `npm run test` and `npm run benchmark`
- [x] tests and/or benchmarks are included
- [ ] documentation is changed or added
- [x] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

File: .taprc [hunk 1]
```diff
@@ -1,4 +1,3 @@
-esm: false
 ts: false
 jsx: false
 flow: false
```
Comment (Eomm) [inline .taprc:3]: adding 

```
check-coverage: false
```

should fix the pipeline

Comment (makuzaverite) [reply .taprc:3]: it didn't

**Comment by climba03003** [comment]
I think there are something wrong with your commit history.
Some of the test case is not updated in your branch.

For example:
https://github.com/fastify/fastify/blob/main/test/async-await.test.js#L425
https://github.com/makuzaverite/fastify/blob/tap15Support/test/async-await.test.js#L425

I think you should reset the commit history to match the upstream before starting a new PR.

File: test/async-await.test.js [hunk 2]
```diff
@@ -576,8 +576,8 @@ test('customErrorHandler support', t => {
     url: '/'
   }, (err, res) => {
     t.error(err)
-    t.strictEqual(res.statusCode, 401)
-    t.deepEqual(
+    t.eqaul(res.statusCode, 401)
```
Comment (climba03003) [inline test/async-await.test.js:579]: Typo
```suggestion
    t.equal(res.statusCode, 401)
```

File: test/async-await.test.js [hunk 3]
```diff
@@ -610,8 +610,8 @@ test('customErrorHandler support without throwing', t => {
...
```
Comment (climba03003) [inline test/async-await.test.js:613]: Typo
```suggestion
    t.equal(res.statusCode, 401)
```

File: test/async-await.test.js [hunk 4]
```diff
@@ -638,8 +638,8 @@ test('customErrorHandler only called if reply not already sent', t => {
     url: '/'
   }, (err, res) => {
     t.error(err)
-    t.strictEqual(res.statusCode, 200)
-    t.deepEqual(
+    t.eqaul(res.statusCode, 200)
```
Comment (climba03003) [inline test/async-await.test.js:641]: Typo
```suggestion
    t.equal(res.statusCode, 200)
```

**Review by climba03003 — CHANGES_REQUESTED** [review]

**Comment by makuzaverite** [comment]
Thanks @climba03003 had a wrong select and replace inside my vscode

File: test/async-await.test.js [hunk 5]
```diff
@@ -117,19 +117,19 @@ test('ignore the result of the promise if reply.send is called beforehand (objec
...
-      t.deepEqual(payload, JSON.parse(body))
-      t.strictEqual(res.statusCode, 200)
+      t.same(payload, JSON.parse(body))
+      t.equal(res.statusCode, 200)
     })
   })
 })
 
-test('server logs an error if reply.send is called and a value is returned via async/await', t => {
+test('server logs an error if reply.send is called and a value equal returned via async/await', t => {
```
Comment (climba03003) [inline test/async-await.test.js:126]: Wrong replacement
```suggestion
test('server logs an error if reply.send is called and a value is returned via async/await', t => {
```

**Review by climba03003 — CHANGES_REQUESTED** [review]
Can you double check it is not missing any of the wrong replacement? I may miss some of them when I glance through this big changes.

File: docs/Hooks.md [hunk 6]
```diff
@@ -434,22 +434,22 @@ Note that the Fastify context in each hook is the same as the plugin where the r
 ```js
 fastify.addHook('onRequest', async function (req, reply) {
   if (req.raw.url === '/nested') {
-    assert.strictEqual(this.foo, 'bar')
+    assert.equal(this.foo, 'bar')
   } else {
-    assert.strictEqual(this.foo, undefined)
+    assert.equal(this.foo, undefined)
```
Comment (mcollina) [inline docs/Hooks.md:439]: This was correct, please restore

File: docs/Hooks.md [hunk 7]
```diff
@@ -434,22 +434,22 @@ Note that the Fastify context in each hook is the same as the plugin where the r
 ```js
 fastify.addHook('onRequest', async function (req, reply) {
   if (req.raw.url === '/nested') {
-    assert.strictEqual(this.foo, 'bar')
+    assert.equal(this.foo, 'bar')
```
Comment (mcollina) [inline docs/Hooks.md:437]: This was correct, please restore

File: docs/Hooks.md [hunk 8]
```diff
@@ -434,22 +434,22 @@ Note that the Fastify context in each hook is the same as the plugin where the r
...
   } else {
-    assert.strictEqual(this.foo, undefined)
+    assert.equal(this.foo, undefined)
   }
 })
 
 fastify.get('/', async function (req, reply) {
-  assert.strictEqual(this.foo, undefined)
+  assert.equal(this.foo, undefined)
```
Comment (mcollina) [inline docs/Hooks.md:444]: This was correct, please restore

File: docs/Hooks.md [hunk 9]
```diff
@@ -434,22 +434,22 @@ Note that the Fastify context in each hook is the same as the plugin where the r
...
   return { hello: 'world' }
 })
 
 fastify.register(async function plugin (fastify, opts) {
   fastify.decorate('foo', 'bar')
 
   fastify.get('/nested', async function (req, reply) {
-    assert.strictEqual(this.foo, 'bar')
+    assert.equal(this.foo, 'bar')
```
Comment (mcollina) [inline docs/Hooks.md:452]: This was correct, please restore

File: test/async-await.test.js [hunk 10]
```diff
@@ -325,13 +325,13 @@ test('inject async await', async t => {
...
     const res = await fastify.inject({ method: 'GET', url: '/' })
-    t.deepEqual({ hello: 'world' }, JSON.parse(res.payload))
+    t.same({ hello: 'world' }, JSON.parse(res.payload))
   } catch (err) {
     t.fail(err)
   }
 })
 
-test('inject async await - when the server is up', async t => {
+test('inject async await - when the server equal up', async t => {
```
[the remark this answers is not in the document]
Comment (zekth) [reply test/async-await.test.js:334]: @makuzaverite still this to fix

**Review by mcollina — APPROVED** [review]
lgtm, thanks for the gargantuan effort!

**Review by zekth — APPROVED** [review]
lgtm huge work

**Pull request merged**